contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,264 +9,282 @@ showTitle: true
 
+On macOS you can just run:
 ```bash
-python3 -m pip install --upgrade pip
+brew install postgresql
 ```
+This installs both the Postgres server and its tools. DO NOT start the server after running this.
 
-then retry the requirements.txt install.
+On Linux you often have separate packages: `postgres` for the tools, `postgres-server` for the server, and `libpostgres-dev` for the `psycopg2` dependencies. Consult your distro's list for an up-to-date list of pacakages.
 
-> **Friendly tip:** If you're running into errors with this step on a Mac with an Apple Silicon chip, you may need to set some environment variables before running the install command, like so:
-    
-```bash
-CFLAGS="-I /opt/homebrew/opt/openssl/include" LDFLAGS="-L /opt/homebrew/opt/openssl/lib" GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1 GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1 pip install -r requirements.txt
-```
-    
-5. If you want to fully test all our features, you'll need to install a few dependencies for SAML to run properly. If you're on macOS run the command below, otherwise check out the official [xmlsec repo](https://github.com/mehcode/python-xmlsec) for more details.
+### 2. Prepare the frontend
 
-```
-brew install libxml2 libxmlsec1 pkg-config
-pip install python3-saml==1.12.0
-```
+1. Install nvm: https://github.com/nvm-sh/nvm. If using fish, you may instead prefer https://github.com/jorgebucaran/nvm.fish.
+
+<blockquote class="warning-note">
+    After installation, make sure to follow the instructions printed in your terminal to add NVM to your <code>$PATH</code>. Otherwise the command line will use your system Node.js version instead.
+</blockquote>
+
+2. Install the latest Node.js 14 (the version used by PostHog in production) with `nvm install 14`. You can start using it in the current shell with `nvm use 14`.
+
+3. Install yarn with `npm install -g yarn@1`.
+
+4. Install Node packages by running `yarn`.
+
+5. Run `yarn typegen:write` to generate types for [Kea](https://kea.js.org/) state management logics used all over the frontend.
+
+> The first time you run typegen, it may get stuck in a loop. If so, cancel the process (`Ctrl+C`), discard all changes in the working directory (`git reset --hard`), and run `yarn typegen:write` again. You may need to discard all changes once more when the second round of type generation completes.
+
+### 3. Prepare the plugin server
 
-6. Install dev requirements:
+1. Assuming Node.js is installed, run `yarn --cwd plugin-server` to install all required packages. We'll run this service in a later step.
+
+### 4. Prepare the Django server
+
+1. Install Python 3.8+. You can do so with Homebrew: `brew install python3`. Make sure to always use `python3` instead of `python`, as the latter may point to Python 2.x on some systems.
```

Comment:
```suggestion
1. Install Python 3.8+. You can do so with Homebrew: `brew install python@3.8` or `brew install python@3.9`.
```

To install a fixed version.

I'd keep `pyenv` because global dependencies can make for confusing bugs but "all the ways to manage python on your machine" can be kept out of scope :)

Reply:
+1 to `pyenv`, makes things easier (at least for me)